LoadingUnloading.lua

```diff
@@ -0,0 +1,506 @@
+			local fillLevels, capacity
+			--normal fillLevels of silo
+			if self.source.getAllFillLevels then 
+				fillLevels, capacity = self.source:getAllFillLevels(g_currentMission:getFarmId())
+			--g_company fillLevels of silo
+			elseif self.source.getAllProvidedFillLevels then --g_company fillLevels
+				--self.managerId should be self.extraParameter!!!
+				fillLevels, capacity = self.source:getAllProvidedFillLevels(g_currentMission:getFarmId(), self.managerId)
+			else
+				return superFunc(self)
+			end
+			local fillableObject = self.validFillableObject
+			local fillUnitIndex = self.validFillableFillUnitIndex
+			local firstFillType = nil
+			local validFillTypIndexes = {}
+			local emptyOnes = 0
+			for fillTypeIndex, fillLevel in pairs(fillLevels) do
+				if self.fillTypes == nil or self.fillTypes[fillTypeIndex] then
+					if fillableObject:getFillUnitAllowsFillType(fillUnitIndex, fillTypeIndex) then
+						for _,data in ipairs(fillTypeData) do
+							--check silo fillLevel
+							if fillLevel > 0 and  fillTypeIndex == data.fillType then
+								--check if specific fillType is reached
+								if not vehicle.cp.driver:isFilledUntilPercantageX(fillTypeIndex,data.maxFillLevel) then 
+									--cover is open, wait till it's open to start load
+									if fillableObject.spec_cover and fillableObject.spec_cover.isDirty then 
+										vehicle.cp.driver:setLoadingState(fillableObject,fillUnitIndex,fillTypeIndex,self)
+										courseplay.debugFormat(2, 'Cover is still opening!')
+										return
+									end
+									--fixes giants bug for Lemken Solitär with has fillunit that keeps on filling to infinity
+									if fillableObject:getFillUnitCapacity(fillUnitIndex) <=0 then 
+										vehicle.cp.driver:resetLoadingState()
+										return
+									else
+									--start loading everthing is ok
+										self:onFillTypeSelection(fillTypeIndex)
+										callback.ok = true
+										return								
+									end
+								else
+									courseplay.debugFormat(2, 'FillLevel reached!')
+									callback.ok = true
+								end
+							else 
+								emptyOnes = emptyOnes+1
+							end
+						end
+					end
+				end
+			end
+			--if all selected fillTypes are empty in the trigger and no fillLevel reached => wait for more
+			if emptyOnes == fillTypeDataSize and not callback.ok then 
+				vehicle.cp.driver:setLoadingState()
+				CpManager:setGlobalInfoText(vehicle, 'FARM_SILO_IS_EMPTY');
+				return
+			end
+		end
+	else 
+		return superFunc(self)
+	end
+end
+LoadTrigger.onActivateObject = Utils.overwrittenFunction(LoadTrigger.onActivateObject,courseplay.onActivateObject)
+
+--LoadTrigger => start/stop driver and close cover once free from trigger
+function courseplay:setIsLoading(superFunc,isLoading, targetObject, fillUnitIndex, fillType, noEventSend)
+	local rootVehicle = self.validFillableObject:getRootVehicle()
+	if courseplay:checkAIDriver(rootVehicle) then
+		if not rootVehicle.cp.driver:is_a(FillableFieldworkAIDriver) then
```

Comment:
see above, it is better to ask the driver if he is ok doing this. Or, consider to implement the whole function in the driver (calling only superfunc in AIDriver and doing what it is doing here in FillableFieldworkAIDriver